parser/src/cfg/mod.rs

```diff
@@ -1766,49 +1766,50 @@ fn parse_mods_held_for_submacro<'a>(
     Ok((mod_keys, unparsed_str))
 }
 
+use phf::phf_map;
+static KEYMODI: phf::Map<&'static str, KeyCode> = phf_map! {
+    "S-" => KeyCode::LShift,
+    "‹⇧" => KeyCode::LShift,
+    "⇧›" => KeyCode::RShift,
+    "C-" => KeyCode::LCtrl,
+    "‹⎈" => KeyCode::LCtrl,
+    "‹⌃" => KeyCode::LCtrl,
+    "⎈›" => KeyCode::RCtrl,
+    "⌃›" => KeyCode::RCtrl,
+    "M-" => KeyCode::LGui,
+    "‹◆" => KeyCode::LGui,
+    "‹⌘" => KeyCode::LGui,
+    "‹❖" => KeyCode::LGui,
+    "◆›" => KeyCode::RGui,
+    "⌘›" => KeyCode::RGui,
+    "❖›"  => KeyCode::RGui,
+    "‹⎇" => KeyCode::LAlt,
+    "A-" => KeyCode::LAlt,
+    "‹⌥" => KeyCode::LAlt,
+    "AG-" => KeyCode::RAlt,
+    "RA-" => KeyCode::RAlt,
+    "⎇›" => KeyCode::RAlt,
+    "⌥›" => KeyCode::RAlt,
+};
+
 /// Parses mod keys like `C-S-`. Returns the `KeyCode`s for the modifiers parsed and the unparsed
 /// text after any parsed modifier prefixes.
 pub fn parse_mod_prefix(mods: &str) -> Result<(Vec<KeyCode>, &str)> {
     let mut key_stack = Vec::new();
     let mut rem = mods;
     loop {
-        if let Some(rest) = rem.strip_prefix("C-") {
-            if key_stack.contains(&KeyCode::LCtrl) {
-                bail!("Redundant \"C-\" in {mods:?}");
-            }
-            key_stack.push(KeyCode::LCtrl);
-            rem = rest;
-        } else if let Some(rest) = rem.strip_prefix("S-") {
-            if key_stack.contains(&KeyCode::LShift) {
-                bail!("Redundant \"S-\" in {mods:?}");
-            }
-            key_stack.push(KeyCode::LShift);
-            rem = rest;
-        } else if let Some(rest) = rem.strip_prefix("AG-") {
-            if key_stack.contains(&KeyCode::RAlt) {
-                bail!("Redundant \"AltGr\" in {mods:?}");
-            }
-            key_stack.push(KeyCode::RAlt);
-            rem = rest;
-        } else if let Some(rest) = rem.strip_prefix("RA-") {
-            if key_stack.contains(&KeyCode::RAlt) {
-                bail!("Redundant \"AltGr\" in {mods:?}");
-            }
-            key_stack.push(KeyCode::RAlt);
-            rem = rest;
-        } else if let Some(rest) = rem.strip_prefix("A-") {
-            if key_stack.contains(&KeyCode::LAlt) {
-                bail!("Redundant \"A-\" in {mods:?}");
-            }
-            key_stack.push(KeyCode::LAlt);
-            rem = rest;
-        } else if let Some(rest) = rem.strip_prefix("M-") {
-            if key_stack.contains(&KeyCode::LGui) {
-                bail!("Redundant \"M-\" in {mods:?}");
+        let mut found_none = true;
+        for (key_s, key_code) in &KEYMODI {
```

Comment:
It doesn't look like this needed to use phf to me. Doesn't this work just as well just being an array/slice of tuples?

Reply:
I wanted to generate the map from two list of keys and prefixes, but then remembered that rust/phf don't support the neccessary const functionality to allow that at compile time, but left if for the future when they do

But currently there is no benefit since the whole list is hardcoded anyway, so can change to an array